astropy/units/physical.py

```diff
@@ -1,133 +1,473 @@
+        if base in ["deg_F", "deg_C", "deg_R"]:
+            base = "K"
+            substitution_was_made = True
+        physical_type_id_components.append((base, power))
+
+    if substitution_was_made:
+        return core.Unit._from_physical_type_id(tuple(physical_type_id_components))
+    else:
+        return unit
+
+
+def _standardize_physical_type_names(physical_type_input):
+    """
+    Convert a string or `set` of strings into a `set` containing
+    string representations of physical types.
+
+    The strings provided in ``physical_type_input`` can each contain
+    multiple physical types that are separated by a regular slash.
+    Underscores are treated as spaces so that variable names could
+    be identical to physical type names.
+    """
+    if isinstance(physical_type_input, str):
+        physical_type_input = {physical_type_input}
+
+    standardized_physical_types = set()
+
+    for ptype_input in physical_type_input:
+        if not isinstance(ptype_input, str):
+            raise ValueError(f"expecting a string, but got {ptype_input}")
+        input_set = set(ptype_input.split("/"))
+        processed_set = {s.strip().replace("_", " ") for s in input_set}
+        standardized_physical_types |= processed_set
+
+    return standardized_physical_types
+
+
+class PhysicalType:
+    """
+    Represents the physical type(s) that are dimensionally compatible
+    with a set of units.
+
+    Instances of this class should be accessed using the ``physical_type``
+    property of `Unit` instances.  This class is not intended to be
+    instantiated directly in user code.
+
+    Parameters
+    ----------
+    unit : u.Unit
+        The unit to be represented by the physical type.
+
+    physical_types : `str` or `set` of `str`
+        A `str` representing the name of the physical type of the unit,
+        or a `set` containing strings that represent one or more names
+        of physical types.
+
+    Notes
+    -----
+    A physical type will be considered equal to an equivalent
+    `PhysicalType` instance (recommended) or a string that contains a
+    name of the physical type.  The latter method is not recommended
+    in packages, as the names of some physical types may change in the
+    future.
+
+    To maintain backwards compatibility, two physical type names may be
+    included in one string if they are separated with a slash (e.g.,
+    ``"momentum/impulse"``).  String representations of physical types
+    may include underscores instead of spaces.
+
+    Examples
+    --------
+    The preferred method to access a physical type is by accessing the
+    ``physical_type`` attribute of a unit.
+
+    >>> import astropy.units as u
+    >>> u.meter.physical_type
+    PhysicalType('length')
+
+    Some units are dimensionally compatible with multiple physical types.
+    A pascal is intended to represent pressure or stress, but the unit
+    decomposition is equivalent to that of energy density.
```

Comment:
and "and stress"

Reply:
Just to confirm that I understood your suggestion correctly, I changed `to represent pressure or stress` to `to represent pressure and stress`.

Reply:
I meant to add it at the end "of energy density and stress" - I don't think Pa is meant to be used for anything but pressue.

Reply:
According to the wikipedia article, Pa is used for stress: https://en.wikipedia.org/wiki/Stress_(mechanics)

Reply:
Yeah, I ended up finding a lot of inconsistency on the World Wide Web™ about what pascals correspond to.  Some sites only said pressure, others said pressure & stress.  I decided to go with the [SI standard](https://www.bipm.org/utils/common/pdf/si-brochure/SI-Brochure-9.pdf), which has the pascal representing "pressure, stress" (pdf page 137).

Reply:
OK, I stand corrected!